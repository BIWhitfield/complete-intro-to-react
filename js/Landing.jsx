// @flow

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import type { RouterHistory } from 'react-router-dom';
import { setSearchTerm } from './actionCreators';

class Landing extends Component {
  props: {
    searchTerm: string,
    handleSearchTermChange: Function,
    history: RouterHistory
  };

  goToSearch = (e: SyntheticEvent) => {
    e.preventDefault();
    this.props.history.push('/search');
  };

  render() {
    return (
      <div className="landing">
        <h1>svideo</h1>
        <form onSubmit={this.goToSearch}>
          <input
            onChange={this.props.handleSearchTermChange}
            value={this.props.searchTerm}
            type="text"
            placeholder="Search"
          />
          <Link to="/search">or Browse All</Link>
        </form>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch: Function) => ({
  handleSearchTermChange: e => {
    dispatch(setSearchTerm(e.target.value));
  }
});

const mapStateToProps = state => ({
  searchTerm: state.searchTerm
});

export default connect(mapStateToProps, mapDispatchToProps)(Landing);
