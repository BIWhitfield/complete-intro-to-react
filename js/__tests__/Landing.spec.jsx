import React from 'react';
import { shallow } from 'enzyme';
import Landing from '../Landing';

describe('Landing', () => {
  it('renders correctly', () => {
    const component = shallow(<Landing />);
    expect(component).toMatchSnapshot();
  });
});
