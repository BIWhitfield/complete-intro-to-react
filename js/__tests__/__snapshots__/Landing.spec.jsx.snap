// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing renders correctly 1`] = `
<div
  className="landing"
>
  <h1>
    svideo
  </h1>
  <input
    placeholder="Search"
    type="text"
  />
  <Link
    replace={false}
    to="/search"
  >
    or Browse All
  </Link>
</div>
`;
